vibin/mediaservers/cxnv2usb.py

```diff
@@ -149,17 +151,19 @@ def __init__(self, prefix: str = "id"):
     def get_id(self, path: _MediaPath) -> MediaId:
         """Returns the generated id for the given path, generating a new one if
         necessary."""
-        if path not in self._by_path:
-            generated = f"{self._prefix}{self._next}"
-            self._next = self._next + 1
-            self._by_path[path] = generated
-            self._by_id[generated] = path
-        return self._by_path[path]
+        with self._lock:
+            if path not in self._by_path:
+                generated = f"{self._prefix}{self._next}"
+                self._next = self._next + 1
+                self._by_path[path] = generated
+                self._by_id[generated] = path
+            return self._by_path[path]
 
     def get_path(self, id: MediaId) -> _MediaPath:
         """Returns the path corresponding to the given id, throwing KeyError
         if none."""
-        return self._by_id[id]
+        with self._lock:
```

Comment:
As an aside: I suspect I don't actually need to acquire the lock here.

I can't find a great reference for Python's threading model, but I suspect that the dictionary lookup would be atomic in practice and, if the path is recorded in the `_by_id` dict, then the rest of the state of `_MediaIdGenerator` would be consistent due to the ordering of operations in the `get_id` method.

Still - it doesn't harm, I can't find a cast iron guarantee that dicts are thread-safe, and if nothing else it's a helpful guard against any future changes that affect the way this object works.

Reply:
Agreed on the dictionary lookup. Your solution here looks good to me.